Bindings/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+
+if(BUILD_PYTHON_WRAPPING)
+    find_package(SWIG 3.0.5 REQUIRED)
+    add_subdirectory(Python)
+endif()
+if(BUILD_PYTHON_WRAPPING)
+    find_package(SWIG 3.0.5 REQUIRED)
+    add_subdirectory(Java)
+endif()
```

Comment:
Oh good point; I hadn't thought of that.

Right now both Java and Python do require other packages, but those are handled in their own directories; should those be moved up here?

Also, if the `if() ... endif()` is here, we should remove them from the Java and Python subdirs so we can unindent all that cmake code; which would be nice :)

As for the duplication, I suggest we address that when we encounter requiring different swig versions for different wrapping.

Nonetheless there is still a bug above; the second `BUILD_PYTHON_WRAPPING` should be `BUILD_JAVA_WRAPPING`.